tests/test_state/test_matcha_state.py

```diff
@@ -1,5 +1,6 @@
 """Tests for Matcha State Service."""
```

Comment:
There isn't any direct testing of the new functions in here (besides one for `test_matcha_state_build_state_from_terraform_output`)  - could you add happy and unhappy path tests for the new functions you've added into `MatchaStateService`?